### PR TITLE
feat: implement audit for experimentation

### PR DESCRIPTION
**Pull request by raji-unni**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: src/controllers/trigger.js [hunk 1]
```diff
@@ -27,6 +28,7 @@ const AUDITS = {
   lhs, // for all lhs variants
   404: notfound,
   'broken-backlinks': backlinks,
+  experiment,
```
Comment (ekremney) [inline src/controllers/trigger.js:31]: better suited term would be `experimentation` for this audit

**Review by ekremney — APPROVED** [review]
LGTM, with one comment:

File: src/controllers/trigger.js (hunk 1 above)
Comment (raji-unni) [reply src/controllers/trigger.js]: Accepted this change. Incorporated.

File: src/controllers/trigger/experiment.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
```
Comment (solaris007) [inline src/controllers/trigger/experiment.js:2]: 2024

File: src/controllers/trigger/experiment.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    AUDIT_REPORT_SLACK_CHANNEL_ID: slackChannelId,
+    SLACK_BOT_TOKEN: token,
+  } = context.env;
+
+  if (!hasText(domainkey) || !hasText(queueUrl)) {
+    throw Error('Required env variables are missing');
+  }
+  const rumApiClient = RUMAPIClient.createFrom(context);
+
+  const urls = await rumApiClient.getDomainList();
```
Comment (solaris007) [inline src/controllers/trigger/experiment.js:37]: as per discussion here we should no longer use the RUM domain list: https://cq-dev.slack.com/archives/C05A45JBP9N/p1706631055605969?thread_ts=1706609179.753669&cid=C05A45JBP9N

but instead use data access layer / getSitesToAudit

Comment (raji-unni) [reply src/controllers/trigger/experiment.js]: updated the code to use data access layer

File: src/controllers/trigger/experiment.js (hunk 2 above)
Comment (raji-unni) [reply src/controllers/trigger/experiment.js]: Incorporated

**Review by ekremney — APPROVED** [review]

**Pull request merged**